src/middlewared/middlewared/plugins/boot.py

```diff
@@ -111,7 +111,26 @@ async def get_state(self):
         """
         Returns the current state of the boot pool, including all vdevs, properties and datasets.
         """
-        return await self.middleware.call('zfs.pool.query', [('name', '=', BOOT_POOL_NAME)], {'get': True})
+        info = await self.middleware.call('zfs.pool.query', [('name', '=', BOOT_POOL_NAME)], {'get': True})
+
+        # WebUI expects this to return the same data as `pool.pool_extend`
+        return {
+            'name': BOOT_POOL_NAME,
+            'status': info['status'],
+            'scan': info['scan'],
+            'topology': await self.middleware.call('pool.transform_topology', info['groups']),
+            'healthy': info['healthy'],
+            'warning': info['warning'],
+            'status_detail': info['status_detail'],
```

Comment:
It's great to have uniform data structures, but it would be even better if we could extract this into a separate method that'll be used both by `pool_extend` and `get_state`; otherwise, we are repeating ourselves here.

Reply:
We'll also be able to use `@returns(Ref('pool_entry'))` here (or `Patch` because we'll have some fields removed, e.g. `id`)

Reply:
Ok... I'll clean this up. Thx @themylogin